ui/src/app/index.tsx

```diff
@@ -1,5 +1,7 @@
 import * as React from 'react';
-import * as ReactDOM from 'react-dom';
 import {App} from './app';
+import {createRoot} from 'react-dom/client';
 
-ReactDOM.render(<App />, document.getElementById('app'));
+const container = document.getElementById('app');
+const root = createRoot(container!);
```

Comment:
```suggestion
const root = createRoot(document.getElementById('app')!);
```

we can probably simplify this, the `container` var is otherwise unused